### PR TITLE
Add 'add_genesis_winner' option to ledger runtime_config

### PR DESCRIPTION
**Pull request by mrmr1993**
This adds the option to override adding the genesis winner to genesis ledgers, and disables it by default for configurations with `proof_level <> Full`. This will be used for integration tests, to ensure that the ledger layout isn't changed.

Checklist:

- [ ] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: src/lib/genesis_ledger_helper/genesis_ledger_helper.ml [hunk 1]
```diff
@@ -280,21 +280,31 @@ module Ledger = struct
...
-          | (_, account) :: _
-            when Public_key.Compressed.equal (Account.public_key account) pk ->
-              accounts
-          | _ ->
-              ( None
-              , Account.create
-                  (Account_id.create pk Token_id.default)
-                  (Currency.Balance.of_int 1000) )
-              :: accounts
+          let add_genesis_winner_account =
```
Comment (psteckler) [inline src/lib/genesis_ledger_helper/genesis_ledger_helper.ml:287]: is it worth reprising the PR comment here, to say that we don't want to change the ledger layout for integration tests?

File: src/lib/genesis_ledger_helper/genesis_ledger_helper.ml [hunk 2]
```diff
@@ -280,21 +280,31 @@ module Ledger = struct
...
+            match accounts with
+            | (_, account) :: _
+              when Public_key.Compressed.equal (Account.public_key account) pk
+              ->
+                accounts
+            | _ ->
+                ( None
+                , Account.create
+                    (Account_id.create pk Token_id.default)
+                    (Currency.Balance.of_int 1000) )
```
Comment (psteckler) [inline src/lib/genesis_ledger_helper/genesis_ledger_helper.ml:305]: what's the significance of the 1000 initial balance here?

File: src/lib/genesis_ledger_helper/genesis_ledger_helper.ml [hunk 3]
```diff
@@ -280,21 +280,31 @@ module Ledger = struct
...
+                :: accounts
+          else accounts
         in
         let accounts =
```
Comment (psteckler) [inline src/lib/genesis_ledger_helper/genesis_ledger_helper.ml:309]: can we name this `accounts_opt`, and the variable on ln 329, `padded_accounts_opt` ?

File: src/lib/genesis_ledger_helper/genesis_ledger_helper.ml (hunk 2 above)
Comment (mrmr1993) [reply src/lib/genesis_ledger_helper/genesis_ledger_helper.ml:313]: It came from [this change](https://github.com/CodaProtocol/coda/pull/4001/files#diff-d0b72396c50f0a47f3245df56b77c428R117), I haven't been able to track down where it originally came from.

**Pull request merged**